jbpm-wb-integration/jbpm-wb-integration-backend/src/main/java/org/jbpm/workbench/wi/backend/server/workitem/ServiceTaskServiceImpl.java

```diff
@@ -203,20 +215,24 @@ public void saveConfiguration(ServiceTasksConfiguration configuration) {
                                    .collect(Collectors.toList()));
                     service.setTriggertitle(widInfo.serviceInfo().trigger().title());
                     
-                    installedServiceTasks.add(service.getName());
                     logger.debug("Adding service task with name {} of type {}", service.getName(), workItem);
-                    repoService.addService(service);
+                    repoService.addService(service, resultMap);
                 } else {
                     logger.warn("Wid annotation was not found on type {}", workItem);
                 }
             }
-            
-            return installedServiceTasks;
+            return resultMap;
         } catch (MalformedURLException e) {
             throw new RuntimeException(e);
         } 
     }
 
+    @Override
+    public String removeServiceTask(ServiceTaskSummary serviceTaskSummary) {
+        repoService.removeServiceTask(serviceTaskSummary.getId(), serviceTaskSummary.getInstalledOn().size() > 0 ? false : true);
```

Comment:
@mswiderski is the logic correct here? Should we disable the ST only when it is used by no project? ....

Reply:
@jstastny-cz, the logic in here https://github.com/kiegroup/jbpm-wb/pull/1399/files#diff-dcc43a73cf2fdc739aaef2b7a4c084a6R138. The UI do never do not invoke the ST, it still being installed by projects. needDisable just let code flexible. if you think, it's needless, we can remove it.